modules/autotagging-commons/common/src/main/java/org/opensearch/rule/SecurityAttribute.java

```diff
@@ -0,0 +1,123 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.rule;
+
+import org.opensearch.core.xcontent.XContentBuilder;
+import org.opensearch.core.xcontent.XContentParseException;
+import org.opensearch.core.xcontent.XContentParser;
+import org.opensearch.rule.autotagging.Attribute;
+
+import java.io.IOException;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import java.util.TreeMap;
+import java.util.stream.Collectors;
+
+/**
+ * Security attribute for the rules. Example:
+ * principal: {
+ *   "username": ["alice", "bob"],
+ *   "role": ["admin"]
+ * }
+ * @opensearch.experimental
+ */
+public enum SecurityAttribute implements Attribute {
+    /**
+     * Represents the principal attribute
+     */
+    PRINCIPAL("principal");
+
+    /**
+     * Represents the username subfield
+     */
+    public static final String USERNAME = "username";
+    /**
+     * Represents the role subfield
+     */
+    public static final String ROLE = "role";
+    private static final TreeMap<Integer, String> PRIORITIZED_SUBFIELDS = new TreeMap<>(Map.of(1, USERNAME, 2, ROLE));
+    private final String name;
```

Comment:
I am slightly unclear as to why `TreeMap` is the right data structure for storing `PRIORITIZED_SUBFIELDS`. Also, if it were right data structure, would have expected to have `TreeMap<String, Integer>` instead of `TreeMap<Integer, String>`